packages/shopify-app-session-storage-prisma/src/prisma.ts

```diff
@@ -38,11 +39,24 @@ export class PrismaSessionStorage<T extends PrismaClient>
 
     const data = this.sessionToRow(session);
 
-    await this.getSessionTable().upsert({
-      where: {id: session.id},
-      update: data,
-      create: data,
-    });
+    try {
+      await this.getSessionTable().upsert({
+        where: {id: session.id},
+        update: data,
+        create: data,
+      });
+    } catch (error) {
+      if (
+        error instanceof Prisma.PrismaClientKnownRequestError &&
+        error.code === 'P2002'
+      ) {
+        console.log(
+          'Caught PrismaClientKnownRequestError P2002, Session already exists',
+        );
+        return true;
```

Comment:
Should we trigger an update here just in case? This is a race condition so quite unlikely that it'll be 2 different concurrent operations, but maybe it can happen?